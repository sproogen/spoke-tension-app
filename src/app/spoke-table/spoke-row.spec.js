import React from 'react'
import { shallow } from 'enzyme'
import SpokeRow from './spoke-row'

jest.mock('../conversions', () => ({
  getTension: jest.fn(() => 100),
}))

const props = {
  spoke: {
    toolId: 'tool',
    spokeId: 'spoke-1.8',
    number: 0,
    reading: '40',
  },
  averageTension: 110,
  updateReading: jest.fn(),
}

describe('SpokeRow', () => {
  describe('actions', () => {
    describe('reading onChange', () => {
      it('should call update reading', () => {
        const wrapper = shallow(<SpokeRow {...props} />)
        const event = {
          preventDefault() { },
          target: { value: '41' },
        }

        wrapper.find('#spoke-reading-0').props().onChange(event)

        expect(props.updateReading).toHaveBeenCalledWith('41')
      })
    })
  })

  describe('render', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<SpokeRow {...props} />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
