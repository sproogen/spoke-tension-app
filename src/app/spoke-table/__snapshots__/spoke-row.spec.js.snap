// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpokeRow render should match snapshot 1`] = `
<WithStyles(TableRow)>
  <WithStyles(TableCell)>
    1
  </WithStyles(TableCell)>
  <WithStyles(TableCell)>
    <TextField
      error={false}
      id="spoke-reading-0"
      margin="normal"
      onChange={[Function]}
      required={false}
      select={false}
      type="number"
      value="40"
      variant="outlined"
    />
  </WithStyles(TableCell)>
  <WithStyles(TableCell)>
    100
  </WithStyles(TableCell)>
  <WithStyles(TableCell)>
    -9
  </WithStyles(TableCell)>
</WithStyles(TableRow)>
`;

exports[`SpokeRow render with no tension should match snapshot 1`] = `
<WithStyles(TableRow)>
  <WithStyles(TableCell)>
    1
  </WithStyles(TableCell)>
  <WithStyles(TableCell)>
    <TextField
      error={true}
      id="spoke-reading-0"
      margin="normal"
      onChange={[Function]}
      required={false}
      select={false}
      type="number"
      value="40"
      variant="outlined"
    />
  </WithStyles(TableCell)>
  <WithStyles(TableCell)>
    -
  </WithStyles(TableCell)>
  <WithStyles(TableCell)>
    -
  </WithStyles(TableCell)>
</WithStyles(TableRow)>
`;
